keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/NpmArtifact.kt

```diff
@@ -14,9 +14,16 @@ data class NpmArtifact(
   override val deliveryConfigName: String? = null,
   override val reference: String = name,
   override val statuses: Set<ArtifactStatus> = emptySet(),
-  override val versioningStrategy: VersioningStrategy = NpmVersioningStrategy,
   override val from: ArtifactOriginFilterSpec? = null
 ) : DeliveryArtifact() {
   override val type = NPM
+
+  override val sortingStrategy: SortingStrategy
```

Comment:
The sorting strategy is auto-selected based on whether the artifact spec is defined in terms of source details or not.